### ISSUE TITLE
Element has no widget of type 'Serenity.IntegerEditor'! .NET Core

### ISSUE THREAD

**Issue by PedroHFranco** [volkanceylan/Serenity#3358]
I'm gotting this erro with .NET Core

Serenity.Web 3.5.5
.NET Core 2.0

> Element has no widget of type 'Serenity.IntegerEditor'! If you have recently changed editor type of a property in a form class, or changed data type in row (which also changes editor type) your script side Form definition might be out of date. Make sure your project builds successfully and transform T4 templates

I'm trying to make an validation when value of select2 change, my select2 is construct by a enum.

My field in row:

```
        [DisplayName("Notificacao"), NotNull, DefaultValue(NotificacaoEvento.Padrao)]
        public NotificacaoEvento? Notificacao
        {
            get { return (NotificacaoEvento?)Fields.Notificacao[this]; }
            set { Fields.Notificacao[this] = (Int32?)value; }
        }
```

My enum cs class:

  ```
  [EnumKey("Cadastros.NotificacaoEvento")]
    public enum NotificacaoEvento
    {
        [Description("Padrão")]
        Padrao = 1,
        [Description("Personalizado")]
        Personalizado = 2
    }
```

My class in form.ts

```
    export interface EventosForm {
        Descricao: Serenity.StringEditor;
        Situacao: Serenity.IntegerEditor;
        AplicativoId: Serenity.LookupEditor;
        ComponenteExtracaoId: Serenity.LookupEditor;
        Notificar: Serenity.IntegerEditor;
        Periodicidade: Serenity.IntegerEditor;
        Notificacao: Serenity.IntegerEditor;
        Assunto: Serenity.StringEditor;
        Corpo: Serenity.StringEditor;
        Usuarios: EventoResponsaveisEditor
    }

    export class EventosForm extends Serenity.PrefixedContext {
        static formKey = 'Cadastros.Eventos';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!EventosForm.init) {
                EventosForm.init = true;

                var s = Serenity;
                var w1 = s.LookupEditor;
                var w2 = s.IntegerEditor;
                var w3 = s.StringEditor;

                Q.initFormType(EventosForm, [
                    'Descricao', w3,
                    'Situacao', w2,
                    'Notificar', w2,
                    'Periodicidade', w2,
                    'Notificacao', w2,
                    'Assunto', w3,
                    'Corpo', w3,
                    'AplicativoId',w1,
                    'ComponenteExtracaoId', w1,
                    'Usuarios', EventoResponsaveisEditor
                ]);
            }
        }
    }
```
This is my Dialog:

    @Serenity.Decorators.registerClass()
    @Serenity.Decorators.panel()
    export class EventosDialog extends Serenity.EntityDialog<EventosRow, any> {
        protected getFormKey() { return EventosForm.formKey; }
        protected getIdProperty() { return EventosRow.idProperty; }
        protected getLocalTextPrefix() { return EventosRow.localTextPrefix; }
        protected getNameProperty() { return EventosRow.nameProperty; }
        protected getService() { return EventosService.baseUrl; }

        protected form: EventosForm;
        constructor() {
            super();

            this.form = new EventosForm(this.idPrefix);

            console.log(this.form);
            this.form.Notificacao.changeSelect2(e => {
                var notificacaoId = Q.toId(this.form.Notificacao.value);
                console.log(notificacaoId);
            });

            this.byId('Assunto').closest('.field').hide();
            this.byId('Corpo').closest('.field').hide();
        }
    }

This is a console.log of my form untill try access Notificacao in dialog:

![image](https://user-images.githubusercontent.com/10779853/37341950-f2a73b1e-26a1-11e8-95fa-42b99e1121dc.png)

When i change my Notificacao to an enum in form, i got this erro:

> Exception: TypeError: Right-hand side of 'instanceof' is not callable at Object.tn [as isAssignableFrom]

Any idea how to solve this ?

Thanks.

**Comment by FinlanStudios** [comment]
I've also been running into this issue since I updated Serenity to 3.5.5. Except, I'm getting this issue with something else. 

`[DisplayName("A&E StyleID"), Column("AeStyleID"), NotNull, ForeignKey("[dbo].[AeStyle]", "AeStyleID"), LeftJoin("AeStyle"), TextualField("A&E StyleID"), LookupInclude]
        [LookupEditor(typeof(AeStyleRow))]
        public Int32? AeStyleID
        {
            get { return Fields.AeStyleID[this]; }
            set { Fields.AeStyleID[this] = value; }
        }`

For some reason after I updated Serenity, this got changed to an IntegerEditor instead of a LookupEditor. I have A LOT of other ID's that are identical to this one, yet this one is the only one that's breaking. I haven't quite figured out why it's changing.

**Comment by FinlanStudios** [comment]
Make sure you have `LookupInclude` on the required fields.

**Comment by FinlanStudios** [comment]
I've also been running into this issue since I updated Serenity to 3.5.5. Except, I'm getting this issue with something else.

[DisplayName("A&E StyleID"), Column("AeStyleID"), NotNull, ForeignKey("[dbo].[AeStyle]", "AeStyleID"), LeftJoin("AeStyle"), TextualField("A&E StyleID"), LookupInclude]
[LookupEditor(typeof(AeStyleRow))]
public Int32? AeStyleID
{
get { return Fields.AeStyleID[this]; }
set { Fields.AeStyleID[this] = value; }
}

For some reason after I updated Serenity, this got changed to an IntegerEditor instead of a LookupEditor in the TS code generated files.

I have A LOT of other ID's that are identical to this one, yet this one is the only one that's breaking. I haven't quite figured out why it's changing.
